write-rules.js

```diff
@@ -0,0 +1,179 @@
+ *
+ * Ensuring that column widths remain readable while viewing the markdown raw.
+ *
+ * If a rule URL can be resolved, the rule will be added with a link to it, otherwise
+ * the rule will be just shown as is.
+ *
+ * Rule configuration is not yet included in the table.
+ *
+ * @param {Object} rules
+ * @returns {String}
+ */
+const createMarkdownTable = (rules) => {
+  const enabledRules = Object.entries(rules)
+    .filter(([, [value]]) => value && value !== 'off')
+    .sort(([ruleA], [ruleB]) => ruleA.localeCompare(ruleB));
+
+  const maxRuleLength =
+    enabledRules.reduce(
+      (maxLength, [rule]) =>
+        rule.length > maxLength ? rule.length : maxLength,
+      0,
+    ) + 7; // adding padding + room for links
+
+  const prettyConfigs = enabledRules.map(([, config]) =>
+    getPrettyConfig(config),
+  );
+
+  const maxConfigLength = prettyConfigs.reduce(
+    (maxLength, [, prettyConfig]) =>
+      prettyConfig.length > maxLength ? prettyConfig.length : maxLength,
+    0,
+  );
+
+  const tableRows = enabledRules.map(([rule], index) => {
+    const ruleBackticks = `\`${rule}\``;
+
+    const url = getUrl(rule);
+
+    const ruleFormatted = url ? `[${ruleBackticks}][${index}]` : ruleBackticks;
+
+    const [severity, prettyConfig] = prettyConfigs[index];
+
+    const columns = [
+      ruleFormatted.padEnd(maxRuleLength + 1, ' '),
+      ('`' + severity + '`').padEnd(8),
+      prettyConfig.padEnd(maxConfigLength, ' '),
+    ];
+
+    const footnote = url && `[${index}]: ${url}`;
+
+    return [`| ${columns.join(' | ')} |`, footnote];
+  });
+
+  const tableHeader = [
+    `| ${'Rule'.padEnd(maxRuleLength, ' ')}  | Severity | ${'Config'.padEnd(
+      maxConfigLength,
+      ' ',
+    )} |`,
+    `| ${''.padEnd(maxRuleLength, '-')}- | -------- | ${''.padEnd(
+      maxConfigLength,
+      '-',
+    )} |`,
+  ].join('\n');
+
+  return `
+
+${tableHeader}
+${tableRows.map(([row]) => row).join('\n')}
+
+[config]: https://github.com/wagtail/eslint-config-wagtail/blob/main/index.js
+${tableRows
+  .map(([, link]) => link)
+  .filter(Boolean)
+  .join('\n')}
+`;
+};
+
+/**
+ * Reads the rules configuration from stylelint and writes an updated table
+ * to the README.md file.
+ *
+ * If run with `--check`, no content will be written and instead an exit
+ * code will fire to flag a CI error.
+ */
+const writeRules = async () => {
+  const eslint = new ESLint();
+  const { rules } = await eslint.calculateConfigForFile('.index.js');
+
+  const markdownTable = createMarkdownTable(rules);
+
+  fs.readFile(README, (fileReadError, buf) => {
+    if (fileReadError) throw fileReadError;
+
+    const content = buf.toString();
+
+    const startOfTable = content.indexOf(COMMENT) + COMMENT.length;
+
+    const newContent = content.substring(0, startOfTable).concat(markdownTable);
+
+    // if in check mode, only compare contents and do not write to the file
+    const args = process.argv.slice(2) || [];
```

Comment:
Looks like a [`no-constant-binary-expression`](https://eslint.org/docs/latest/rules/no-constant-binary-expression)? 😄

Reply:
Haha. Oh the irony. Thanks for fixing it.